src/components/menu/menu-item/menu-item.component.js

```diff
@@ -95,14 +121,20 @@ const MenuItem = ({
     className: classes,
     href,
     target,
-    onClick,
+    onClick: onClick || isChildSearch.current ? updateFocusOnClick : undefined,
```

Comment:
```suggestion
    onClick: onClick || (isChildSearch.current ? updateFocusOnClick : undefined),
```

Just to make the expression a bit more clear